### PR TITLE
Fixes #12065 - refresh hosts in invocation detail

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: app/assets/javascripts/template_invocation.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ function job_invocation_form_binds() {
 }
 
 function delayed_refresh(url, data){
+
```
Comment (ohadlevy) [inline app/assets/javascripts/template_invocation.js]: nitpick: whitepsace

File: app/assets/javascripts/template_invocation.js [hunk 2]
```diff
@@ -57,3 +58,16 @@ function delayed_refresh(url, data){
...
+  return {
+    hosts_needs_refresh: $("div#hosts").data('needs-refresh'),
+    host_ids_needing_status_update: fetch_ids_of_hosts('status'),
+    host_ids_needing_provider_update: fetch_ids_of_hosts('provider'),
+    host_ids_needing_actions_update: fetch_ids_of_hosts('actions')
+  }
+}
+
+function fetch_ids_of_hosts(attribute){
+  return _.map($('div#hosts td.host_' + attribute + '[data-needs-refresh="true"]'), function(elem) { return $(elem).data('id') });
```
Comment (ohadlevy) [inline app/assets/javascripts/template_invocation.js]: nitpick: maybe refresh_require=true?

File: app/views/job_invocations/show.js.erb [hunk 3]
```diff
@@ -1,6 +1,22 @@
 $('div#status_chart').html('<%=j job_invocation_chart(@job_invocation) %>');
 $('div#status').flot_pie();
 
+<% if params[:hosts_needs_refresh] == 'true' && @job_invocation.resolved? %>
+  var hosts_tab = $('div#hosts')
+  hosts_tab.html('<%=j render('tab_hosts', :job_invocation => @job_invocation, :hosts => @hosts) %>');
+  hosts_tab.data('needs-refresh', '');
```
Comment (ohadlevy) [inline app/views/job_invocations/show.js.erb]: I dont know why, but i really dislike show.js.erb :)

Comment (ares) [inline app/views/job_invocations/show.js.erb]: do you know any better way that does not duplicate the view logic? the only alternative that I can imagine is to render some template html and send just data from server to client, then use the template to clone html and fill in the data but 
  1. we'd duplicate at least translation data or logic like "status 1 means Failed with cross icon"
  2. I saw too many issues with this approach in existing Foreman forms (compute profiles, interface, parameters)

Probably best would be to use some templating language that can be used in both ruby na js (e.g. [mustache](https://mustache.github.io/)) but I don't want to introduce another templating solution :-)

**Comment by ares** [comment]
tests failed because of some dependency installation issue on 1.9.3

**Comment by iNecas** [comment]
It replaces #51, right?

**Comment by ares** [comment]
correct

**Comment by iNecas** [comment]
The build issue should go away now https://github.com/theforeman/foreman/pull/2786

**Comment by iNecas** [comment]
Tested and works nicely, ACK

**Comment by ares** [comment]
Thanks, merging

**Pull request merged**